libvast/src/system/consensus.cpp

```diff
@@ -618,8 +621,8 @@ make_install_snapshot(Actor* self, peer_state& peer) {
   req.last_snapshot_index = peer.last_snapshot_index;
   req.byte_offset = peer.snapshot->size() - peer.snapshot->in_avail();
   // Construct at most chunks of 1 MB.
-  auto one_mb = std::streamsize{1 << 20};
-  req.data.resize(std::min(one_mb, peer.snapshot->in_avail()));
+  auto min = detail::narrow_cast<std::streamsize>(1_MiB);
+  req.data.resize(std::min(min, peer.snapshot->in_avail()));
```

Comment:
Very confusing. `std::streamsize` is an unspecified **signed** type, while `size_type` is very much unsigned, so what is the reason for this 2 way double conversion?

Reply:
The reason is that `std::min` requires types of the same kind. So you have to cast either the first or the second argument to the type of the other one. I could have also casted `std::stream_size` to `size_t`, but since 1MiB is never negative, I opted for this route.

Reply:
Well, now you call `vector::resize()` with a signed type, which can actually be negative [according to cppreference](https://en.cppreference.com/w/cpp/io/basic_streambuf/in_avail).

Reply:
I think this is the better option 👍